docs/topics/request-response.rst

```diff
@@ -177,18 +177,48 @@ Request objects
         :meth:`replace`.
 
     .. attribute:: Request.meta
+       :value: {}
 
-        A dict that contains arbitrary metadata for this request. This dict is
-        empty for new Requests, and is usually  populated by different Scrapy
-        components (extensions, middlewares, etc). So the data contained in this
-        dict depends on the extensions you have enabled.
+        A dictionary of arbitrary metadata for the request.
 
-        See :ref:`topics-request-meta` for a list of special meta keys
-        recognized by Scrapy.
+        You may extend request metadata as you see fit.
 
-        This dict is :doc:`shallow copied <library/copy>` when the request is
-        cloned using the ``copy()`` or ``replace()`` methods, and can also be
-        accessed, in your spider, from the ``response.meta`` attribute.
+        Request metadata can also be accessed through the
+        :attr:`~scrapy.http.Response.meta` attribute of a response.
+
+        To pass data from one spider callback to another, consider using
+        :attr:`cb_kwargs` instead.
+
+        To keep some data across 3 or more spider callbacks, however, request
+        metadata may be the right choice.
```

Comment:
Hm, I think cb_kwargs also works ok in this case - you can pass something like `partial_data` dict or list as a kwarg to all callbacks

Reply:
The use case I am trying to describe here is where you pass some specific metadata key to almost every callback, and new future callbacks are likely to get it as well, but often most callbacks only pass it over to the next callback. So, for example, `start_url` if you want to keep track of the root URL that originated every callback.

In those cases, adding the same parameter to every callback, specially to callbacks whose only function is to pass the parameter over to the next callback, may be too verbose, and using meta instead (and eventually `STICKY_META_KEYS`, https://github.com/scrapy/scrapy/pull/4141, which is what inspired this paragraph) may be better.

I do need to come out with a better wording. Let me give it at least 1 more try.